crates/providers/lattice-controller/src/lib.rs

```diff
@@ -0,0 +1,630 @@
+    ) -> ProviderInvocationResult<()> {
+        let client = self
+            .get_connections()
+            .await
+            .get_client(&arg.lattice_id)
+            .await?;
+        let mut hm = HashMap::new();
+        if let Some(ref c) = arg.credentials {
+            for (k, v) in c {
+                hm.insert(
+                    k.to_string(),
+                    interface_client::RegistryCredential {
+                        password: v.password.clone(),
+                        token: v.token.clone(),
+                        username: v.username.clone(),
+                        registry_type: "".to_string(),
+                    },
+                );
+            }
+        }
+
+        client
+            .put_registries(hm)
+            .await
+            .map_err(|e| ProviderInvocationError::Provider(e.to_string()))?;
+        Ok(())
+    }
+
+    /// Auction a provider on the lattice
+    #[instrument(level = "debug", skip(self, _ctx), fields(actor_id = ?_ctx.actor, lattice_id = ?arg.lattice_id))]
+    async fn auction_provider(
+        &self,
+        _ctx: Context,
+        arg: ProviderAuctionRequest,
+    ) -> ProviderInvocationResult<Vec<ProviderAuctionAck>> {
+        Ok(self
+            .get_connections()
+            .await
+            .get_client(&arg.lattice_id)
+            .await?
+            .perform_provider_auction(&arg.provider_ref, &arg.link_name, arg.constraints.clone())
+            .await
+            .map(|v| {
+                v.into_iter()
+                    .map(|a| ProviderAuctionAck {
+                        host_id: a.host_id,
+                        link_name: a.link_name,
+                        provider_ref: a.provider_ref,
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .map_err(|e| ProviderInvocationError::Provider(e.to_string()))?)
+    }
+
+    /// Auction an actor on the lattice
+    #[instrument(level = "debug", skip(self, _ctx), fields(actor_id = ?_ctx.actor, lattice_id = ?arg.lattice_id))]
+    async fn auction_actor(
+        &self,
+        _ctx: Context,
+        arg: ActorAuctionRequest,
+    ) -> ProviderInvocationResult<Vec<ActorAuctionAck>> {
+        Ok(self
+            .get_connections()
+            .await
+            .get_client(&arg.lattice_id)
+            .await?
+            .perform_actor_auction(&arg.actor_ref, arg.constraints.clone())
+            .await
+            .map(|v| {
+                v.into_iter()
+                    .map(|a| ActorAuctionAck {
+                        actor_ref: a.actor_ref,
+                        host_id: a.host_id,
+                    })
+                    .collect::<Vec<_>>()
+            })
+            .map_err(|e| ProviderInvocationError::Provider(e.to_string()))?)
+    }
+
+    /// Retrieve all hosts on the lattice
+    #[instrument(level = "debug", skip(self, _ctx, arg), fields(actor_id = ?_ctx.actor, lattice_id = %arg.to_string()))]
```

Comment:
Nit: Change to `skip_all`

Reply:
- [ ] Fix `skip_all` for `instrument` call